### ISSUE TITLE
Installing Cheat Engine doesn't get any easier than this.

### ISSUE THREAD

**Issue by fade2gray** [cheat-engine/cheat-engine#2099]
#### If you have ever installed a game like Minecraft, or any other game which installs an anti-cheat program like Badlion, you will have to disable/remove the anti-cheat.

If you still have problems with the initial download, try #1954.

https://user-images.githubusercontent.com/1332009/158638001-9aff3939-95de-479c-a0a9-723932c0f036.mp4

**Comment by PEKADRAGON** [comment]
i still cant its says d1tvmj3dyb0q0v.cloudfront.net

**Comment by fade2gray** [comment]
#1605
#1954

**Comment by PEKADRAGON** [comment]
![1647450067152](https://user-images.githubusercontent.com/101735671/158646330-6999777e-d414-49bd-8faa-353dc1675d52.jpg)

**Comment by bapaa** [comment]
the code execution cannot proceed because hhctrl.ocx was not found. Reinstalling the program may fix this problem help

**Comment by polarpup** [comment]
i cant intsall it it says could not call proc.

**Comment by fade2gray** [comment]
@polarpup See [cheat engine "could not call proc"](https://www.google.co.uk/search?q=cheat+engine+%22could+not+call+proc%22&newwindow=1&source=hp&ei=rb5uYqn1HoybgQap5bHwBg&iflsig=AJiK0e8AAAAAYm7MvWbaYbeQL0Y9AKpJbYuAIKEkyubv&ved=0ahUKEwjp_4uo5b73AhWMTcAKHalyDG4Q4dUDCAg&oq=cheat+engine+%22could+not+call+proc%22&gs_lcp=Cgdnd3Mtd2l6EAwyBAgAEB46CwguEIAEELEDEIMBOggILhCxAxCDAToLCAAQgAQQsQMQgwE6EQguEIAEELEDEIMBEMcBEKMCOhQILhCABBCxAxCDARDHARCjAhDUAjoOCC4QgAQQsQMQxwEQ0QM6BQgAEIAEOggIABCxAxCDAToICC4QgAQQsQM6CAgAEIAEELEDOhEILhCABBCxAxCDARDHARDRAzoRCC4QgAQQsQMQgwEQxwEQrwE6CwgAEIAEELEDEMkDOgUIABCSAzoLCC4QgAQQsQMQ1AI6CAguEIAEENQCOgsILhCABBDHARCvAToHCAAQgAQQClAAWMlIYJRnaABwAHgAgAGEAYgB6AiSAQM2LjaYAQCgAQGgAQI&sclient=gws-wiz)

**Comment by davivmcd** [comment]
> #### This video was made on a clean Windows 10 virtual machine and the only security/antivirus/firewall software installed is that installed by Microsoft Windows.
> #### Some common problems encountered when downloading the installer from the CE website.
> 1. You must stay connected to the internet when running the installer, otherwise installation cannot complete.
> 2. If you use a VPN or proxy, you may need to temporarily disable them.
> 3. If you have installed any third party security/firewall software, you may have to temporarily disable/uninstall it (if you use Avast antivirus/firewall, see [How to temporarily disable Avast Antivirus and Firewall #2104](https://github.com/cheat-engine/cheat-engine/issues/2104)).
> 4. If you still can't download the installer from the CE website, it could be due to your service provider blocking it or even your browser's ad blocker (see [this](https://github.com/cheat-engine/cheat-engine/issues/2113#issuecomment-1079455348)) – you can also try downloading the installer for the latest release from the repo [here](https://github.com/cheat-engine/cheat-engine/releases).
> 
> #### Issues you may encounter when try to run CE after successfully downloading and installing it.
> 1. There are some programs/games that will write registry entries that can block CE from running, and you may see an error message like…
>    > 'C:\Program Files\Cheat Engine 7.4\Cheat Engine.exe'.
>    > Make Sure you typed the name correctly, and then try again...
>    
>    
>    Running `windowsrepair.exe` from inside the CE installation directory might fix this.
> 2. **_ANTI-CHEAT PROGRAMS!!!_**
>    If CE doesn't run at all after installation, it may be because you have installed a game like Minecraft or some other game that installs an anti-cheat program, such as Badlion. If that is the case, you will have to disable/remove the anti-cheat (read [this](https://github.com/cheat-engine/cheat-engine/issues/1310#issuecomment-671395570) on how to remove Badlion).
> 
> ## See [Google Translate](https://translate.google.co.uk/?hl=en&tab=wT&sl=en&tl=pl&text=This%20video%20was%20made%20on%20a%20clean%20Windows%2010%20virtual%20machine%20and%20the%20only%20security%2Fantivirus%2Ffirewall%20software%20installed%20is%20that%20installed%20by%20Microsoft%20Windows.%0A%0ASome%20common%20problems%20encountered%20when%20downloading%20the%20installer%20from%20the%20CE%20website%3A%0A%0A1.%20You%20must%20stay%20connected%20to%20the%20internet%20when%20running%20the%20installer%2C%20otherwise%20installation%20cannot%20complete.%0A%0A2.%20If%20you%20use%20a%20VPN%20or%20proxy%2C%20you%20may%20need%20to%20temporarily%20disable%20them.%0A%0A3.%20If%20you%20have%20installed%20any%20third%20party%20security%2Ffirewall%20software%2C%20you%20may%20have%20to%20temporarily%20disable%2Funinstall%20it%20(if%20you%20use%20Avast%20antivirus%2Ffirewall%2C%20see%20https%3A%2F%2Fgithub.com%2Fcheat-engine%2Fcheat-engine%2Fissues%2F2104).%0A%0A4.%20If%20you%20still%20can%27t%20download%20the%20installer%20from%20the%20CE%20website%2C%20it%20could%20be%20due%20to%20your%20service%20provider%20blocking%20it%20or%20even%20your%20browser%27s%20ad%20blocker%20(see%20%5Bthis%5D(https%3A%2F%2Fgithub.com%2Fcheat-engine%2Fcheat-engine%2Fissues%2F2113%23issuecomment-1079455348))%20%E2%80%93%20you%20can%20also%20try%20downloading%20the%20latest%20release%20from%20the%20repo%20%5Bhere%5D(https%3A%2F%2Fgithub.com%2Fcheat-engine%2Fcheat-engine%2Freleases).%0A%0AIssues%20you%20may%20encounter%20when%20try%20to%20run%20CE%20after%20successfully%20downloading%20and%20installing%20it%3A%0A%0A1.%20There%20are%20some%20programs%2Fgames%20that%20will%20write%20registry%20entries%20that%20can%20block%20CE%20from%20running%2C%20and%20you%20may%20see%20an%20error%20message%20like%E2%80%A6%0A%0A%22C%3A%5CProgram%20Files%5CCheat%20Engine%207.4%5CCheat%20Engine.exe.%20%0AMake%20Sure%20you%20typed%20the%20name%20correctly%2C%20and%20then%20try%20again...%22%0A%0ARunning%20%60windowsrepair.exe%60%20from%20inside%20the%20CE%20installation%20directory%20might%20fix%20this.%0A%0A2.%20**_ANTI-CHEAT%20PROGRAMS!!!_**%20If%20CE%20doesn%27t%20run%20at%20all%20after%20installation%2C%20it%20may%20be%20because%20you%20have%20installed%20a%20game%20like%20Minecraft%20or%20some%20other%20game%20that%20installs%20an%20anti-cheat%20program%2C%20such%20as%20Badlion.%20If%20that%20is%20the%20case%2C%20you%20will%20have%20to%20disable%2Fremove%20the%20anti-cheat%20(read%20https%3A%2F%2Fgithub.com%2Fcheat-engine%2Fcheat-engine%2Fissues%2F1310%23issuecomment-671395570%20on%20how%20to%20remove%20Badlion).%0A%0AIf%20you%20still%20have%20problems%20downloading%20the%20installer%2C%20try%20https%3A%2F%2Fgithub.com%2Fcheat-engine%2Fcheat-engine%2Fissues%2F1954&op=translate)
> If you still have problems downloading the installer, try #1954.
> 
>  Cheat.Engine.-.Easy.Download.and.Install.mp4

The current installer for 7.4 is blocked by Windows Defender:
![image](https://user-images.githubusercontent.com/105137635/167279237-e3403cef-cdbc-4189-8675-3e5397aa2b8b.png)

This was run from a fresh Windows 10 VM with the latest Microsoft security updates applied.

**Comment by cheat-engine** [comment]
did you check out the video ? Including the part where runtime protection gets disabled ?

**Comment by KushGene** [comment]
Disabling VPN did it for me. Thanks!

**Comment by HiroHitosis** [comment]
how do i fix this it say's "Site cannot be reached"

**Comment by fade2gray** [comment]
@Solev3 See https://github.com/cheat-engine/cheat-engine/issues/2165

**Comment by lazyangustc82** [comment]
windows defender tells me , that the installer has threat:
PUADlManager:Win32/OfferCore.

**Comment by wilianwrech** [comment]
This is pure bullshit, this installer is installing some malware, I downloaded the installer and SKIPPED the adware and I got the same message "... potentially unwanted software.". Cheat Engine was not installed at all, but something was because now my anti-virus is finding a TROJAN and it is not able to remove it.

**Comment by cheat-engine** [comment]
Your anti virus is going of because it's detecting cheat engine itself

**Comment by wilianwrech** [comment]
> Your anti virus is going of because it's detecting cheat engine itself

If it is detecting Cheat Engine itself, if I take another computer build Cheat Engine myself, and then copy this build to the computer with this anti-virus, it should detect as a virus, which is not the case.

**Comment by cheat-engine** [comment]
inno setup with cheat engine combined and the cheat engine signature does trigger it

**Comment by davivmcd** [comment]
> did you check out the video ? Including the part where runtime protection gets disabled ? it's at 0:40

Just so I'm clear,

The solution to this version being detected by Windows Defender as a virus, is to disable Windows Defender? So you're basically admitting that this version is a virus.

As a programmer, are you actually satisfied by that answer? I'm not sure if you've been working on the project since the beginning or just tagged on recently, but ensuring CheatEngine isn't flagged as malicious software by the majority of AV suites on the market is kind of an important detail.

Until this issue is resolved (or assurances that a plan to resolve this issue is in effect), I will not be contributing anymore towards the project's Patreon.

**Comment by cheat-engine** [comment]
Nothing to be done about it
https://www.microsoft.com/en-us/wdsi/threats/malware-encyclopedia-description?Name=HackTool:Win32/CheatEngine!MSR&ThreatID=2147742984
Cheat Engine is a "hacktool" so they will always 'detect' it no matter what

**Comment by yoll123** [comment]
![image](https://user-images.githubusercontent.com/108743526/177348331-4aafd872-6ac5-4610-aaf3-b7d3d2cd6b71.png)
It doesn't work

**Comment by Adrenesis** [comment]
> As for patreon, the version there is detected as well so you wouldn't like it anyhow (of course there you do have my 100% guarantee it only installs CE even if you click next without reading)

What kind of threat is that?

The purpose of it being Open Source is to guarantee that provided binaries are equal to builded binaries. The whole paradigme of linux's pakage installer is to compare provided and builded binairies, and this is part of the open source paradigm.

There is absolutly no way that the open source aspect of the package authorize you to provide viruses, third party softwares or malwares for you own money, **this doesn't exist in any philosophical space of Free Libre and Open Source Software (FLOSS); it's just trash.**

I mean you could go to the higher trash tier like some other open source software, where you pay for binaries or you build them, but somehow the documentation is not maintained, incomplete or just complete trash. But you choosed the sub-trash tier by providing software that **should not exist in the first place and which are part of how we build bot armies to mine crypto-coins, to spread ransomware or even to steal information on users**.

CE in that state is not Libre nor Open Source, **it is just a way to trick your users** into thinking they have that cool Open Source package, that is clean and all, but which is a highway for malwares and this is morally condamnable to highest point as a legit Open Source devs for the past 10 years.

I don't get any money, but I don't spread viruses, and when **Open Source is a goal for the sanity of us all**, a patreon, an aggressive donation popup, the logo of sponsored company at the start of the app is widely enough (**just look at godot engine**), you can have all of those at the same time if you want. 

I mean you disassemble executables, right? **so you know better than everyone that when you provide obscure third party package it is wrong, it is a wrong way to think**. I would even prefer CheatEngine to have it's own crypto-miner in a freeware package but made by you, be sure of that. **Because right now, what you are doing is giving keys to obscure companies to mess with people computers.**

**CE could be educational**, but right now, it is not, it's just a Trojan, as an app, as a name, as a "brand", and **probably the coolest way for noobs to learn anything at all, but how to bloat a computer.**

I have nothing to add, just think about it, and please, have mercy for us, and recover from this wrong way to think Open Source, because **this such a pain to find some disassembly and ram debugger tool that we can _trust_**

**Comment by arslaniftikhar143** [comment]
For those who can't install CE.
allow the cheat engine from browser and app control of windows defender. Then you are ready to go :D.
![image](https://user-images.githubusercontent.com/89269453/178101283-05ea949a-2111-4048-9064-21c838a764c6.png)

**Comment by ghramsey** [comment]
Having same problem a everyone else. Windows defender blocks the program from being run. Telling it to "unblock" not only does not work. Defender arbitrarily deleted the installer without recovery ie it didn't place it into the recycle bin it deleted it fully.    Windows defender cannot be disabled anymore in most recent version of windows.    And there's no running windows repair inside CE.  It won't let me open CE so that's not an option.

**Comment by fade2gray** [comment]
@ghramsey 
> Windows defender cannot be disabled anymore in most recent version of windows.

Rubbish.
https://support.microsoft.com/en-us/windows/turn-off-defender-antivirus-protection-in-windows-security-99e6004f-c54c-8509-773c-a4d776b77960

**Comment by ghramsey** [comment]
> @ghramsey
> 
> > Windows defender cannot be disabled anymore in most recent version of windows.
> 
> Rubbish. https://support.microsoft.com/en-us/windows/turn-off-defender-antivirus-protection-in-windows-security-99e6004f-c54c-8509-773c-a4d776b77960   I found that option after I made the post but the setting would not stick while it detected 'PuP/malware'.  I understand why it does that, but it's a pain the ass to fix.
Had to dismiss the warning(s) before it would allow the setting change.

I should clarify too, this was only for the installer. I had a previous version of CE v7.3 running fine, but wanted the update.    The installer was being flagged by defender I think perhaps b/c it offers other software during the install process.  I was able to make it install, but it was a headache.

Once installed CE ran OK following use of windowsrepair file.

**Comment by BoffinBrain** [comment]
> You must stay connected to the internet when running the installer, otherwise installation cannot complete.

But why? What are you uploading/downloading?

**Comment by fade2gray** [comment]
@BoffinBrain 
The initial download from the main CE website is just a link to the full installer hosted on another site.

See https://github.com/cheat-engine/cheat-engine/issues/2292#issuecomment-1200402172

**Comment by BoffinBrain** [comment]
In my opinion, the old offline installer was far better. No need to give unnecessary internet access privileges to an installer, especially when it's now known that there are multiple layers of questionable adware in it. The fact that people are reporting issues when using a VPN or PiHole basically confirms that the installer is communicating with untrusted sources. Also, the secondary payload downloaded from the internet automatically means we don't get to inspect the checksum or digital signatures.

**Comment by fade2gray** [comment]
@BoffinBrain 
The way I see it, the installer earns the author a few bucks in advertising - would you begrudge them that? Besides, you can just decline the adware, and providing you know how to deal with your security software settings, you get what you want for free with just a little effort and minimal inconvenience. Other than that, anyone can learn to compile their own from source.

**Comment by BoffinBrain** [comment]
I don't mind that. My issue is mainly about having an installer, which has just been given **administrator rights**, downloading any and all kinds of **unverified payloads** from **unknown remote sites** (which could themselves be compromised or altered without prior notice) and all without the user's consent/knowledge.

When doing things like this, additional risk factors need to be considered, such as, what if a remote advertising partner gets compromised? How do you ensure compliance with the user's local privacy laws when requesting these ads? Does the installer even verify the digital signature of what it's installing on the user's machine? You see, it gets messy real fast.

This would _not_ be an issue with an installer that had pre-bundled sponsored links and the like, which was commonplace in the last decade.

**Comment by fade2gray** [comment]
Well, you are more savvy regarding that aspect of the issue, would it not be better to create a new issue addressing your concerns with the installer?

**Comment by rondlite** [comment]
Not affiliated but artmoney works and has no virus warnings.. 
It is 2022 you can edit memory without triggering virus warning false or not.

**Comment by sonicex123** [comment]
Hi, when I use Cheat Engine 7.4 appear a error, it seems to generate some file that occupies my C drive slot, making his space 0.
the error message seems to say that the file is stored in a certain path but I can't find it, where can I find these files?

**Comment by fade2gray** [comment]
@BoffinBrain 

> Does the installer even verify the digital signature of what it's installing on the user's machine?

See https://youtu.be/v7QsS5qp0og?t=90

**Comment by vrecusko** [comment]
dark byte does not pack malware into the CE setup. he just uses an installer that has ad space. its not surprising that sketchy companies will try to spread their scams there. its just an easy way to get revenue for dark byte, and an easy way for sketchy companies to catch an unsuspecting user, who spam-clicks the continue button without being cautious.

**Comment by protosam** [comment]
It's unfortunate that the victims of this type of malware are kids and technologically under educated.

**Comment by Maahil12** [comment]
Yo ngl I need it for Android but I can't find it so pls help

**Comment by cxtal** [comment]
Peak cringe this one, but ok...

> The purpose of it being Open Source is to guarantee that provided binaries are equal to builded binaries. 

You are delirious. There is no guarantee that the same code compiles to the same binary twice unless you use a compiler that does so. 

> The whole paradigm of linux's pakage installer is to compare provided and builded binairies, and this is part of the open source paradigm.

Outright insane. What is "provided" and what is "builded" and whom'st've doing the compares, man. Wtf...

> There is absolutly no way that the open source aspect of the package authorize you to provide viruses, 

Which package? OS refers to code, not packages. There is no OS authorization process, Kevin.

>  third party softwares or malwares for you own money, this doesn't exist

The GNU/GPL license is a legal virus. Sure, your self-serving morals could perhaps argue that it is a good thing...

> I mean you could go to the higher trash tier like some other open source software, where you pay for binaries or you build them, 

Some enterprise software is "trash tier" because, you, that can barely form a coherent sentence, let alone other technicalities and lack of knowledge, says so?

> as someone pointed out, CE is not open source

You can see the source code. What do you want? You want something else.

> it's unfair to noobs

Why would someone that cannot operate an antivirus (or a spelling checker, not that it would save the day in your case) even use this software? What makes you feel so entitled? @cheat-engine is someone that lays out their work and you are whining because you cannot click the "Skip All" button. 

What's wrong with the 90s? I mean, for real, why did you just mention the 90s? The 80s weren't great either. Are you 12? Wait, don't answer that.

> I have nothing to add

*goes on to add another blotch of text*

 > sentence form of "If you pay on my patreon, I can guarantee that you will not get bloated", makes CE looks more like

Look, Kevin. I have to be really honest with you here. That response should be understood by you (as it is understood by others that are laughing at you) that if you are unable to click the "Skip All" button when the installer asks to install other software, then maybe you could pay the creator and they might just agree to give you some special cookie treatment (compared to the rest of us that noticed the "Skip" and "Skip All" buttons).

**Comment by cxtal** [comment]
> The fact that people are reporting issues when using a VPN or PiHole basically confirms that the installer is communicating with untrusted sources. 

So, why do you trust your VPN provider, who "untrusted" the sources and for what reason? Your VPN is not an anti-virus (it is a  tool that sends all your traffic to some company, that you barely know, such that your straight friends won't tell your mom that you're gay when they find out your IP and also somehow trace it back to your house).

> Also, the secondary payload downloaded from the internet automatically means we don't get to inspect the checksum or digital signatures.

That's what checksums and digital signatures are for. To determine the origin and authenticity of the software downloaded from the Internet.

Weird.

**Comment by cxtal** [comment]
> Not affiliated but artmoney works and has no virus warnings.. It is 2022 you can edit memory without triggering virus warning false or not.

There were computers in the 80s that had no memory protections whatsoever and barely any anti-virus that were very far from stuff like NX. What made you think that year 2022 made writing to memory less restrictive? 

In other news, you should pop your collar to match the glasses. Otherwise, I know this might not seem reasonable to you, but please advertise your software somewhere else. It is just impolite.

**Comment by cxtal** [comment]
> It's unfortunate that the victims of this type of malware are kids

Karen, perhaps you should educate your kids not to cheat.

> and technologically under educated.

We have nothing to fear, but fear itself... and the technologically under-educated.

**Comment by cxtal** [comment]
> Sorry to inform you, Adware is considered Malware, and is Dark Byte choice to do so, no one else's.

Did you ever manage to forgive yourself for informing that person that Adware is considered Malware or is it still gnawing at your very soul?

**Comment by protosam** [comment]
> Karen, perhaps you should educate your kids not to cheat.

It's always great to see people's cognitive dissonance shine through.

**Comment by protosam** [comment]
@cxtal I've been pretty busy, but fortune shines upon the squeaky wheel. I quit my job this week and have some time to give this repository my undivided attention.

**Comment by Artoria2e5** [comment]
I *think* Windows defender and other antiviruses are not triggering on CE: the installers get triggered while the installed things don't. To me it looks more like the bundle installer making it mad. I have 7.5 triggering on "PUADlManager:Win32/OfferCore" and well... It does try to push me into installing some "RAV" thing with an "Accept" button. Which is the literal definition of "potentially unwanted software".

Older Windows defender had a different "App:Cheat_Engine_BundleInstaller" that, despite its name, erroneously targets actual installed CE binaries. This is not the one popping up on me today.

**Comment by cheat-engine** [comment]
There is also a decline button on the left side

but yes with 7.5 i have moved a lot of stuff that was detected as dangerous to compressed cepack files so they don't trigger av's (32 bit tutorial, driver, trainermaker files)

**Comment by soupgalore** [comment]
Hello I have a problem when I try to use it on LDplayer I keep changing the values but at some point all the values dissapear so there is 0 results i dont know what to do can someone help?

**Comment by soupgalore** [comment]
hello I have two problems that keep happening when I try to use it on LDPlayer it doesnt work. once I keep changing the values and use next scan sometimes the thing goes blank and there is no values found. Another thing wrong is when I try to open it with Physical Memory it says "You are missing the driver. Try reinstalling cheat engine, and try to disable your antivirus before doing so." can someone please help?

**Comment by Arkevorkhat1** [comment]
@cheat-engine Yeah, because bundling software totally doesn't completely ruin your reputation, obviously.
There's a reason *reputable* developers don't do it, and that's because no reputable company will pay you to bundle their software, so you're stuck with the scummy ones who don't understand anything beyond "MORE INSTALL MORE BETTER". There's no way for any of your users to know that any of this software is legitimate, or even that it has any relation to the thing your installer claims it to be. 

I get you wanna make a quick buck, but sinking your own ship isn't the way to do it.

**Comment by fade2gray** [comment]
@Arkevorkhat1 https://github.com/cheat-engine/cheat-engine/issues/2621#issuecomment-1606053810

**Comment by VA1DER** [comment]
Version 7.5 lights up VirusTotal like a Christmas tree.  Some adware detections.  Some PUA detections.  But also trojans, and generic malware detections.

Since I  have zero intention of saying yes to any of the ads in the installer, is there a safer version of the installer available?

**Comment by cheat-engine** [comment]
Yes, the patreon version has less detections on virustotal: https://www.patreon.com/cheatengine

**Comment by VA1DER** [comment]
The [mostly benign detections](https://www.virustotal.com/gui/file/3b99af2905d9a7c49570bb8d39877cee7890d8674cea2af1edcc482bb87dbbb6/relations) from the binaries that building this project creates is far less concerning than the [much more serious detections](https://www.virustotal.com/gui/file/6d0e14d66da69c163f824f8fa7d87de3eea41cdbd48c0973de296cf6d2d0fed3/detection) coming from the installers.  The community score for the above installer on VirusTotal is -81.  This is not the kind of score you see attached to something benign.  There are also some alarming sandbox reports in the comments.

I'm not sure you know (or **can** know) what the actual potential real-world threat is from the what you've included, since some of those downloaders change their payloads day to day.  The people behind them often engage in bait and switch tactics even with the vendors they persuade to carry them.

**It is recommend to all to use one of the many [forks](https://github.com/cheat-engine/cheat-engine/forks) which offer untainted builds.  The https://github.com/Zomby7e/pure-cheat-engine fork seems clean.**

Patreon is for rewarding and supporting artists and content creators for great content.  A decision on Patreon support will be made **after** the software is used and assessed.  That decision will be based solely on the merits of this software once I can (safely) use and assess those merits.
